cypress/utils/results.js

```diff
@@ -255,35 +256,41 @@ const receiveResultsEmail = () => {
 
 const receiveResultsSms = () => {
   const phone = "0600000000"
-  cy.intercept(
-    {
-      method: "POST",
-      url: "/api/simulation/*/followup",
-    },
-    {
-      statusCode: 200,
-      body: {
-        surveyOptin: true,
-        phone: "0600000000",
-      },
-    }
-  ).as("post-receive-results-sms")
-
   cy.get("[data-testid='send-email-and-sms-button']", {
     timeout: 20000,
   })
     .should("be.visible")
     .click()
   // scroll to input#phone
-  cy.get("input#phone").scrollIntoView().should("be.visible").type(phone)
-  cy.get(".fr-btn:contains(J'accepte d'être recontacté ou recontactée)")
-    .should("be.visible")
-    .click()
-
-  cy.wait("@post-receive-results-sms").should(({ request, response }) => {
-    expect(request.method).to.equal("POST")
-    expect(response.statusCode).to.equal(200)
-  })
+  const ABTestingEnvironment = storageService.local.getItem("ABTesting") || {}
+  if (ABTestingEnvironment.Followup_SMS.value === "show") {
+    cy.intercept(
+      {
+        method: "POST",
+        url: "/api/simulation/*/followup",
+      },
+      {
+        statusCode: 200,
+        body: {
+          surveyOptin: true,
+          phone: "0600000000",
+        },
+      }
+    ).as("post-receive-results-sms")
+    cy.get("input#phone").scrollIntoView().should("be.visible").type(phone)
+    cy.get(".fr-btn:contains(J'accepte d'être recontacté ou recontactée)")
+      .should("be.visible")
+      .click()
+    cy.wait("@post-receive-results-sms").should(({ request, response }) => {
+      expect(request.method).to.equal("POST")
+      expect(response.statusCode).to.equal(200)
+    })
+  } else if (ABTestingEnvironment.Followup_SMS.value === "hide") {
+    cy.get("input#phone").should("not.exist")
+    cy.get(".fr-btn:contains(J'accepte d'être recontacté ou recontactée)").type(
+      "{esc}"
+    )
+  }
```

Comment:
J'aurais tendance à éviter de mettre des conditions d'AB-test là où on peut s'en passer, surtout dans la CI qui est stable car on fixe les versions d'AB-test. 
Ici on aura temporairement uniquement la version sans téléphone donc seule la ligne `cy.get("input#phone").should("not.exist")` nous intéresse sur la CI

Reply:
Si je fais juste not.exist, la modale reste active et fait planter le test. 
J'ai pensé le développement en mode SMS activé ou désactivé en me basant sur l'AB testing mais il faudrait se baser sur la config SMS et vérifier si elle est active ou non.
Je te propose donc une constante au début de la fonction en mode SMS = false et remplacer mes conditions avec la variable.
Tu en penses quoi ?

Reply:
![image](https://github.com/betagouv/aides-jeunes/assets/18016058/50a307b0-b8bf-4c08-a987-6257cbc00740)

J'ai pas de problème de mon côté en faisant ma suggestion dans `receiveResultsEmail` :

```js
const receiveResultsEmail = () => {
  cy.intercept({
    method: "POST",
    url: "/api/simulation/*/followup",
  }).as("post-receive-results-email")

  cy.get("[data-testid='send-email-and-sms-button']", {
    timeout: 20000,
  })
    .should("be.visible")
    .click()

  const email = "prenom.nom@beta.gouv.fr"
  cy.get("input#phone").should("not.exist")
  cy.get("input#email").should("be.visible").type(email)
  cy.get(".fr-btn:contains(J'accepte d'être recontacté ou recontactée)")
    .should("be.visible")
    .click()

  cy.wait("@post-receive-results-email").should(({ request, response }) => {
    expect(request.method).to.equal("POST")
    expect(response.statusCode).to.equal(200)
  })

  cy.get('[data-testid="simulation-id"')
    .invoke("text")
    .then((simulationId) => {
      cy.task("getLastEmail", email)
        .its("headers.subject")
        .should("includes", simulationId)
    })
}
```

Reply:
Je n'ai pas eu de soucis avec la fonction email et donc je n'ai pas fait de modification, je peux rajouter le not.exist dans cette fonction mais il faudra la supprimer par la suite.
Pour moi le vrai test se fait dans la fonction `receiveResultsSms` , je pense que tu n'as pas pu le voir en local, car la fonction de vérification des mails ne fonctionne pas en local et stop le déroulement des tests.

Reply:
> il faudra la supprimer par la suite.

Oui, c'est juste temporaire, ce sera à revert

> Pour moi le vrai test se fait dans la fonction receiveResultsSms

Oui, tu peux le mettre dans cette méthode, c'est juste que dans le cas de cette modification, nous n'avons plus de résultats par SMS, donc le nom n'est plus cohérent

> la fonction de vérification des mails ne fonctionne pas en local et stop le déroulement des tests.

Yes, j'ai ajouté un ticket pour tenter de corriger ça https://trello.com/c/ZYMTi25C/1690-faire-passer-les-tests-e2e-en-local